src/components/Tracks/Track.js

```diff
@@ -1,24 +1,33 @@
-import {
-  ListItem,
-  ListItemButton,
-  Typography,
-} from "@mui/material";
+import { ListItem, ListItemButton, Typography } from "@mui/material";
 import React from "react";
+import { useCallback } from "react";
+import { useCurrentTrackDispatch } from "../../store/track-context";
+
+function Track({ track, selected }) {
+  const setCurrentTrack = useCurrentTrackDispatch();
+
+  const handleClick = useCallback(() => {
+    setCurrentTrack(track);
```

Comment:
When a track is clicked, it becomes the current one. So, <Track> component is responsible for dispatching.